PYME/LMVis/Extras/kinetics_recipes.py

```diff
@@ -0,0 +1,127 @@
+
+            else:
+                print('HDBScan installed, clustering with that')
+                clusterer = kin_rcps.HDBSCANClustering(recipe, input_name='filtered_coalesced',
+                                                     search_radius=script_init.clusteringRadius,
+                                                     min_clump_size=script_init.minimumClusterSize,
+                                                     clump_column_name=script_init.clusterColumnName,
+                                                     output_name='clusters')
+
+            recipe.add_module(clusterer)
+
+            recipe.add_module(FilterTable(recipe, inputName='clusters', outputName='filt_clusters',
+                                                    filters={script_init.clusterColumnName: [0.0, sys.maxsize]}))
+
+            recipe.add_module(kin_rcps.FindBlinkStateDurations(recipe, inputName='filt_clusters', outputName='unfiltered_kinetics',
+                                                             labelKey=script_init.clusterColumnName, onTimesColName=script_init.onTimesColumnName,
+                                                             offTimesColName=script_init.offTimesColumnName))
+
+            recipe.add_module(FilterTable(recipe, inputName='unfiltered_kinetics', outputName='filtered_kinetics',
+                                          filters={script_init.offTimesColumnName: [0.0, sys.maxsize],
+                                                   script_init.onTimesColumnName: [minblink, sys.maxsize]}))
+
+            recipe.add_module(kin_rcps.FitSwitchingRates(recipe, inputName='filtered_kinetics', outputName='kinetics_out',
+                                                       onTimesColName=script_init.onTimesColumnName,
+                                                       offTimesColName=script_init.offTimesColumnName,
+                                                       minOffTimeSecondsToFit=script_init.minimumOffTimeInSecondsToFit,
+                                                       maxOffTimeSecondsToFit=script_init.maximumOffTimeInSecondsToFit,
+                                                       offTimesFitBinSizeInSeconds=script_init.offTimeFitBinWidthInSeconds))
+            recipe.execute()
+            self.visFr.pipeline.selectDataSource('filtered_kinetics')
+
+        print 'RUNTIME OF ' + str(time()-t0)
+
+class ExportEmphist(object):
+    def __init__(self, visFr):
+        self.visFr = visFr
+
+        visFr.AddMenuItem('Dye Kinetics', 'Save empirical histogram JSON file to disk', self.exp_emphist, helpText='')
+
+    def exp_emphist(self, event):
+
+        self.pipeline = self.visFr.pipeline
+
+        recipe = self.visFr.pipeline.recipe
+
+        json_exporter = kin_rcps.save_EmpHistJson(recipe, inputName='emphist_data', outputName='emphist_file', outputdir='C:\\Users\\bdr25\\Desktop\\data\\JSON files')
```

Comment:
check out `PYME.IO.FileUtils.nameUtils` to default to things which will be sane on other peoples computers